site/content/posts/2022-06-01-json-logging-ldap-ui.md

```diff
@@ -0,0 +1,91 @@
+
+With this release, the Pinniped CLI logs have also been enhanced to emit useful information such as timestamps and line numbers of files, as shown in the example below:
+
+```
+$ pinniped get kubeconfig
+Tue, 24 May 2022 11:18:50 EDT  cmd/kubeconfig.go:584  discovered CredentialIssuer  {"name": "concierge-config"}
+Tue, 24 May 2022 11:18:50 EDT  cmd/kubeconfig.go:419  discovered Concierge operating in impersonation proxy mode
+Tue, 24 May 2022 11:18:50 EDT  cmd/kubeconfig.go:432  discovered Concierge endpoint  {"endpoint": "https://abcd"}
+Tue, 24 May 2022 11:18:50 EDT  cmd/kubeconfig.go:447  discovered Concierge certificate authority bundle  {"roots": 1}
+Tue, 24 May 2022 11:18:50 EDT  cmd/kubeconfig.go:469  discovered WebhookAuthenticator  {"name": "wa"}
+```
+
+
+## LDAP User Interface
+
+With [v0.18.0](https://github.com/vmware-tanzu/pinniped/releases/tag/v0.18.0) we are also adding support for a User Interface that users can access to input their credentials and login to their LDAP or ActiveDirectory Identity Provider(IDP). This feature is a first step in our effort to provide a UI-driven workflow for our users. We will have more features that support UI workflows coming up in our next release, so stay tuned for that!
+
+When using the Pinniped CLI, a successful login takes the user to the regular form_post success page, just like the previously supported *browser authcode flow* for an OIDC IDP.
+This feature changes how the `pinniped get kubeconfig` cli deals with ambiguous flows. Previously, if there was more than one flow advertised for an IDP, the cli would require users to use the `--upstream-identity-provider-flow` flag.  Now, it chooses the first flow type in the Supervisor's discovery response by default.
+
+Here's a snapshot of the UI hosted by the Pinniped Supervisor:
+![LDAP / ActiveDirectory UI for login ](/docs/img/ldap-ad-ui.png)
+
+### Features included in the UI  
+The UI will provide the following features for now:
+
+* A username and password fields along with the submit button
+* Generalized error messaging for failed logins that do not expose sensitive information  
+* Provides information easily allowing a user to identify the screen as belonging to Pinniped with the name of their Identity Provider displayed on the page
+* Addresses basic security concerns for web forms such as use of HTTPS, a password field, CSRF protection and redirect protection
+* Prevents LDAP injection attacks
+* Screens that are accessible and friendly to screen readers
+* Screens that are friendly to password managers
+
+### Future enhancements for the UI
+This is our effort to provide a basic UI for logins. We will enhance the UI further in future releases based on the feedback we receive from our users.
+This can include (but is not limited to ) the following features:
+* A rich client with the use of javascript
```

Comment:
This isn't really a feature in itself... We might need it in the future to support other features, but I don't think it belongs on this list.

Reply:
Removed it. Thanks.